src/main/proto/th2_grpc_data_provider/data_provider.proto

```diff
@@ -301,24 +235,45 @@ message MessageSearchRequest {
 
   google.protobuf.Int32Value result_count_limit = 4; // Sets the maximum amount of messages to return. By default it is absent (unlimited)
 
-  StringList stream = 5; // Sets the stream ids to search in. Case-sensitive. Required.
+  repeated MessageStream stream = 5; // Sets the stream names and directions to search in. Case-sensitive. Required.
   google.protobuf.BoolValue keep_open = 6; //  If the search has reached the current moment, it is necessary to wait further for the appearance of new data. Set by default to false.
 
-  repeated MessageID resume_from_ids = 7; // List of message IDs to restore the search. If given, it has the highest priority and ignores 'stream' (uses streams from the ids), 'start_timestamp' and 'resume_fromId'. By default it is absent.
+  repeated MessageStreamPointer stream_pointer = 7; // List of stream pointers to restore the search from. start_timestamp will be ignored if this parameter is specified. By default it is absent.
 
-  repeated Filter filters = 8; // List of message search filters. (See 'search/sse/message' HTTP method https://github.com/th2-net/th2-rpt-data-provider)
+  repeated Filter filter = 8; // List of message search filters. (See 'search/sse/message' HTTP method https://github.com/th2-net/th2-rpt-data-provider)
   google.protobuf.BoolValue attached_events = 9; // If true, it will additionally load attachedEventsIds. Set by default to false.
-  google.protobuf.Int32Value lookup_limit_days = 10; // The number of days that will be viewed on the first request to get the one closest to the specified timestamp. By default it is absent - not limited to the past and up to the present moment to the future.
 }
 
-/* The object returned by the search function (searchMessages/searchEvents) */
-
-message StreamResponse {
+message MessageSearchResponse {
   oneof data {
-    EventData event = 1;
-    MessageData message = 2;
-    EventMetadata event_metadata = 3;
-    LastScannedObjectInfo last_scanned_object = 4;
-    StreamsInfo stream_info = 5;
+    MessageGroupResponse message = 1;
+    MessageStreamPointers message_stream_pointers = 2;
   }
 }
+
+message MessageStreamPointers {
+  repeated MessageStreamPointer message_stream_pointer = 1;
+}
+
+/* This object contains the data of the last message in the stream on which
+ * the search has stopped, it is necessary to continue the search from the same point */
+ message MessageStreamPointer {
+  MessageStream message_stream = 1;
+  bool has_started = 2;
+  bool has_ended = 3;

```

Comment:
What do these parameters mean?

Reply:
These flags indicate the state of the stream. Has_started means there is data that matches our request. Has_ended means all of the relevant data was already processed and there is nothing left.